tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -65,6 +71,155 @@ void check_and_context_switch() {
     // proceed
 }
 
+/**
```

Comment:
Do we need to copy every noc api to a `eth_noc` version? I would prefer if we only made the copy where we needed to add `risc_context_switch()` for functionality. That way if things improve in the noc api, we benefit here as well.

Reply:
Makes sense, will update

Reply:
Updated as per discussion to include the regular dataflow api, so we do not need to duplicate the functions in this file, only the ones which need context switch are added.